src/modules/mc_pos_control/mc_pos_control_main.cpp

```diff
@@ -787,8 +837,8 @@ MulticopterPositionControl::limit_pos_sp_offset()
 	pos_sp_offs.zero();
 
 	if (_control_mode.flag_control_position_enabled) {
-		pos_sp_offs(0) = (_pos_sp(0) - _pos(0)) / _params.sp_offs_max(0);
-		pos_sp_offs(1) = (_pos_sp(1) - _pos(1)) / _params.sp_offs_max(1);
+		/* Limit maximum setpoint offset in position in body frame */
+		scale_with_body_fixed_gains(pos_sp_offs, _pos_sp - _pos, _params.sp_offs_max.einv(), false);
```

Comment:
Yes, definitely makes sense. However i do not see how this changes with this PR. The same issue exists on current master. The division has just been packed into a function here. Same with L1218...